addon-resizer/nanny/nanny_lib.go

```diff
@@ -114,16 +124,24 @@ func PollAPIServer(k8s KubernetesClient, est ResourceEstimator, pollPeriod time.
 		estimation := est.scaleWithNodes(num)
 
 		// If there's a difference, go ahead and set the new values.
-		overwrite := shouldOverwriteResources(estimation, resources.Limits, resources.Requests)
+		overwrite, op := shouldOverwriteResources(estimation, resources.Limits, resources.Requests)
 		if overwrite == nil {
 			log.V(4).Infof("Resources are within the expected limits. Actual: %+v, accepted range: %+v", *resources, estimation.AcceptableRange)
 			continue
 		}
 
+		now := time.Now()
+		if (op == scaleDown && now.Before(lastChange.Add(scaleDownDelay))) ||
```

Comment:
Can you point me to the specific line in test where this thing is properly unit-tested. I don't see a line where we check what happens if last change is slightly before or slightly after scale****Delay. But maybe I'm missing something.